apps/src/templates/HintsDisplay.jsx

```diff
@@ -0,0 +1,60 @@
+module.exports = React.createClass({
+
+  propTypes: {
+    hintReviewTitle: React.PropTypes.string.isRequired,
+    seenHints: React.PropTypes.array.isRequired,
+    unseenHints: React.PropTypes.array.isRequired,
+    onUserViewedHint: React.PropTypes.func.isRequired,
+    lightbulbSVG: React.PropTypes.node.isRequired,
+  },
+
+  getInitialState: function () {
+    return {
+      showNextUnseenHint: false
+    };
+  },
+
+  viewHint: function () {
+    this.props.onUserViewedHint();
+    this.setState({
+      showNextUnseenHint: true
+    });
+  },
+
+  render: function () {
+
```

Comment:
I don't have a strong feeling either way about it, but I use it in larger functions to distinguish broad-scoping 'function initialization' stuff (config parsing and whatnot) from just jumping into the body of the function. So, 

```javascript
function (options) {
  var options = options || {};

  // start the body of the function
}
```

Versus
```javascript
function (options) {

  // first self-contained bit of logic

  // second self-contained bit of logic

  // and so on
}
```

But it's mostly aesthetics plus an old habit, and it does violate our styleguide.